superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets.tsx

```diff
@@ -0,0 +1,198 @@
+import { generateFiltersSetId } from './utils';
+import { Filter } from '../types';
+
+const FilterSet = styled.div`
+  display: grid;
+  align-items: center;
+  justify-content: center;
+  grid-template-columns: 1fr;
+  grid-gap: 10px;
+  padding-top: 10px;
+`;
+
+const StyledTitle = styled.h4`
+  width: 100%;
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+  color: ${({ theme }) => theme.colors.grayscale.dark1};
+  margin: 0;
+  overflow-wrap: break-word;
+
+  & > .ant-select {
+    width: 100%;
+  }
+`;
+
+const ActionButtons = styled.div`
+  display: grid;
+  flex-direction: row;
+  grid-template-columns: 1fr 1fr;
+  ${({ theme }) =>
+    `padding: 0 ${theme.gridUnit * 2}px ${theme.gridUnit * 2}px`};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+
+  .btn {
+    flex: 1;
+  }
+`;
+
+const Sets = styled(ActionButtons)`
+  grid-template-columns: 1fr;
+`;
+
+type FilterSetsProps = {
+  dataMaskState: DataMaskUnitWithId;
+  onFilterSelectionChange: (
+    filter: Pick<Filter, 'id'> & Partial<Filter>,
+    dataMask: Partial<DataMaskState>,
+  ) => void;
+};
+
+const FilterSets: React.FC<FilterSetsProps> = ({
+  onFilterSelectionChange,
+  dataMaskState,
+}) => {
+  const dispatch = useDispatch();
+  const [filtersSetName, setFiltersSetName] = useState('');
+  const filterSets = useSelector<any, FilterSets>(
+    state => state.nativeFilters.filterSets ?? {},
+  );
+  const filterSetsConfigs = useSelector<any, FiltersSet[]>(
+    state => state.dashboardInfo?.metadata?.filter_sets_configuration || [],
+  );
+  const [selectedFiltersSetId, setSelectedFiltersSetId] = useState<
+    string | null
+  >(null);
+
+  const takeFiltersSet = (value: string) => {
+    setSelectedFiltersSetId(value);
+    if (!value) {
+      return;
+    }
+    const filtersSet = filterSets[value];
+    Object.values(filtersSet.dataMask?.nativeFilters ?? []).forEach(
+      dataMask => {
+        const { extraFormData, currentState, id } = dataMask as MaskWithId;
+        onFilterSelectionChange(
+          { id },
+          { nativeFilters: { extraFormData, currentState } },
+        );
+      },
+    );
+  };
+
+  const handleSaveFilterSets = () => {
+    dispatch(
+      setFilterSetsConfiguration(
+        filterSetsConfigs.concat([
+          {
+            name: filtersSetName.trim(),
+            id: generateFiltersSetId(),
+            dataMask: {
+              nativeFilters: dataMaskState,
+            },
+          },
+        ]),
+      ),
+    );
+    setFiltersSetName('');
+  };
+
+  const handleDeleteFilterSets = () => {
+    dispatch(
+      setFilterSetsConfiguration(
+        filterSetsConfigs.filter(
+          filtersSet => filtersSet.id !== selectedFiltersSetId,
+        ),
+      ),
+    );
+    setFiltersSetName('');
+    setSelectedFiltersSetId(null);
+  };
+
+  return (
+    <Sets>
+      <FilterSet>
+        <StyledTitle>
+          <div>{t('Choose filters set')}</div>
```

Comment:
I think this should be "Choose filter set"

Reply:
and same here, because this functionality going to be changed